### PR TITLE
EUI-6646 Restricted case access user attempts to access restricted case

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-7382
https://tools.hmcts.net/jira/browse/EUI-6646

### Change description ###
EUI-7382 Restricted Case Access - Tech Tasks
EUI-6646 Restricted Case Access - HMCTS Case worker, JOH or Professional users attempts to access restricted case (no access)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/ccd-case-ui-toolkit",
-  "version": "5.0.50-sscs-sit-release",
+  "version": "5.0.49-restricted-case-access-v2",
```
Comment (andywilkinshmcts) [inline package.json:3]: toolkit version has gone backwards - is this definitely what you want

File: projects/ccd-case-ui-toolkit/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/ccd-case-ui-toolkit",
-  "version": "5.0.50-sscs-sit-release",
+  "version": "5.0.49-restricted-case-access-v2",
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/package.json:3]: Toolkit version has gone backwards

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts [hunk 3]
```diff
@@ -85,22 +85,27 @@ export class CaseResolver implements Resolve<CaseView> {
...
-  private checkAuthorizationError(error: any) {
+  private checkAuthorizationError(error: any, caseReference: string) {
+    console.error(error);
     // TODO Should be logged to remote logging infrastructure
     if (error.status === 400) {
       this.router.navigate(['/search/noresults']);
       return of(null);
     }
-    console.error(error);
+    // Error 403, navigate to restricted case access page
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts:99]: I think I would move this to after the CaseResolver.EVENT_REGEX block below. I don't imagine that an event trigger would return 403 very often, but if it did then you'd trigger the restricted case logic, which may not be what you want

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Couple of points to address - concern about CCD toolkit version, and the possibly move the retricted cases block to after the check for event triggers

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts [hunk 4]
```diff
@@ -289,6 +289,34 @@ describe('CaseResolver', () => {
...
+        email: 'testing2@mail.com',
+        active: true,
+        roles: ['caseworker-ia-caseofficer']
+      };
+      sessionStorageService.getItem.and.returnValue(JSON.stringify(userInfo));
+
+      caseResolver = new CaseResolver(caseNotifier, draftService, navigationNotifierService, router, sessionStorageService);
+      caseResolver
+        .resolve(route)
+        .then(data => {
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts:312]: @johnbenjamin-hmcts - minor formatting; for single-line functions you can remove the `{` and `}`

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts [hunk 5]
```diff
@@ -289,6 +289,34 @@ describe('CaseResolver', () => {
...
+          expect(data).toBeFalsy();
+        }, err => {
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts:314]: @johnbenjamin-hmcts - same comment as above on minor formatting

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts [hunk 6]
```diff
@@ -85,22 +85,27 @@ export class CaseResolver implements Resolve<CaseView> {
...
           return this.caseNotifier.cachedCaseView;
         }),
-        catchError(error => this.checkAuthorizationError(error))
+        catchError(error => this.checkAuthorizationError(error, cid))
       ).toPromise();
   }
 
-  private checkAuthorizationError(error: any) {
+  private checkAuthorizationError(error: any, caseReference: string) {
+    console.error(error);
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts:93]: @johnbenjamin-hmcts - please remove this `console.error()` statement if no longer required

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - some very minor observations; otherwise OK

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts]: Thank you @Daniel-Lam, I have done the changes as you suggested.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts (hunk 5 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.spec.ts]: Thank you @Daniel-Lam, I have done the changes as you suggested.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts (hunk 6 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts]: @Daniel-Lam I removed this `console.error` statement.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts:102]: Thank you @andywilkinshmcts, I have made the changes as you suggested.

File: projects/ccd-case-ui-toolkit/package.json (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/package.json]: I have updated the version.

File: package.json (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply package.json]: I have updated the version.

**Review by Daniel-Lam — APPROVED** [review]

**Review by andywilkinshmcts — DISMISSED** [review]

**Pull request closed without merging**